Workflow/Sources/Workflow.swift

```diff
@@ -49,37 +49,71 @@
 /// }
 /// ```
 ///
-public protocol Workflow: AnyWorkflowConvertible {
-    /// Defines the state that is managed by this workflow.
-    associatedtype State
+#if swift(>=5.7)
+    public protocol Workflow<Rendering, Output>: AnyWorkflowConvertible {
```

Comment:
for the time being, decided to just make this change on `Workflow` itself, as i believe that is by far the most useful change. if there are other compelling additions we could add them in the future.

with this change, we can now leverage the following forms of type erasure:

```swift
func boxedAnyWorkflow() -> AnyWorkflow<Rendering, Output> {
  MyWorkflow().asAnyWorkflow()
}

func existentialAnyWorkflow() -> any Workflow<Rendering, Output> {
  if someCondition {
    return MyWorkflow()
  } else {
    return MyOtherWorkflow()
  }
}

func opaqueWorkflow() -> some Workflow<Rendering, Output> {
  MyWorkflow()
}
```